spec/GraphQLOverHTTP.md

```diff
@@ -187,94 +305,298 @@ This request could be sent via an HTTP GET as follows:
 
+GET requests MUST NOT be used for executing mutation operations. If the values
+of {query} and {operationName} indicate that a mutation operation is to be
+executed, the server MUST respond with error status code `405` (Method Not
+Allowed) and halt execution. This restriction is necessary to conform with the
+long-established semantics of safe methods within HTTP.
 
 ## POST
 
-A GraphQL POST request instructs the server to perform a query or mutation operation. A GraphQL POST request MUST have a body which contains values of the request parameters encoded according to the value of `Content-Type` header of the request.
+A GraphQL POST request instructs the server to perform a query or mutation
+operation. A GraphQL POST request MUST have a body which contains values of the
+request parameters encoded in one of the officially recognized GraphQL media
+types, or another media type supported by the server.
 
-A client MUST include a `Content-Type` with a POST request.
+A client SHOULD indicate the media type of a request body using the
+`Content-Type` header as specified in
+[RFC7231](https://datatracker.ietf.org/doc/html/rfc7231).
+
+Note: If the client does not include a `Content-Type` header then
+`application/json` will be assumed.
+
+### Legacy watershed
+
+Before 1st January 2025 (`2025-01-01T00:00:00Z`), if a client does not know the
+media types the server supports then it SHOULD use
+`Content-Type: application/json` in requests. After this date, if a client does
+not know the media types the server supports then it SHOULD use
+`Content-Type: application/graphql+json` in requests.
 
 ### Example
 
-Given a POST request with a content type of `application/graphql+json` here is an example request body:
+If we wanted to execute the following GraphQL query:
+
+```graphql example
+query ($id: ID!) {
+  user(id: $id) {
+    name
+  }
+}
+```
+
+With the following query variables:
+
+```json example
+{
+  "id": "QVBJcy5ndXJ1"
+}
+```
+
+This request could be sent via an HTTP POST to the relevant URL using the JSON
+encoding with the headers:
+
+```headers example
+Content-Type: application/graphql+json
+Accept: application/graphql+json
+```
+
+And the body:
 
 ```json example
 {
-  "query": "query($id: ID!){user(id:$id){name}}",
+  "query": "query ($id: ID!) { user(id:$id) { name } }",
   "variables": { "id": "QVBJcy5ndXJ1" }
 }
 ```
 
 # Response
 
-When a GraphQL server receives a request, it must return a well‐formed response. The server's
-response describes the result of executing the requested operation if successful, and describes
-any errors encountered during the request.
+When a server receives a GraphQL request, it must return a well‐formed response.
+The server's response describes the result of validating and executing the
+requested operation if successful, and describes any errors encountered during
+the request.
+
+A server must comply with
+[RFC7231](https://datatracker.ietf.org/doc/html/rfc7231).
+
+## Interpreting the request
```

Comment:
Yeah; you might be right. I moved this around quite a few times before putting it here; I think I was thinking of it as "client" and "server" rather than "request" and "response" but the sections are clearly request/response so I should move it.

Reply:
Moved into the "Media type" section instead.